internal/storage/caching/fast_stat_bucket.go

```diff
@@ -253,9 +285,13 @@ func (b *fastStatBucket) ListObjects(
 		return
 	}
 
-	// Note anything we found.
-	b.insertMultiple(listing.Objects)
+	if b.BucketType() == gcs.Hierarchical {
+		b.insertHierarchicalListing(listing)
+		return
+	}
 
+	// note anything we found.
```

Comment:
Please fix the comment. Original comment was good already.

Reply:
noted, will update in subsequent PRs as dont want to block this change on it.